cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4MeasureDefBuilder.java

```diff
@@ -262,27 +239,102 @@ private void validateImprovementNotationCode(CodeableConcept improvementNotation
         }
     }
 
-    private CodeableConcept getGroupImprovementNotationExt(MeasureGroupComponent group) {
-        var ext = group.getExtensionByUrl(MEASUREREPORT_IMPROVEMENT_NOTATION_EXTENSION)
-                .getValue();
-        assert ext instanceof CodeableConcept;
-        return (CodeableConcept) ext;
+    public CodeDef getMeasureBasis(Measure measure) {
+
+        var ext = measure.getExtensionByUrl(MeasureConstants.POPULATION_BASIS_URL);
+        // check for population-basis Extension, assume boolean if no Extension is found
+        if (ext != null) {
+            return makeCodeDefFromExtension(ext);
+        }
+        return null;
     }
 
-    private boolean groupHasImprovementNotationExt(MeasureGroupComponent group) {
-        return group.getExtensionByUrl(MEASUREREPORT_IMPROVEMENT_NOTATION_EXTENSION) != null;
+    private CodeDef makeCodeDefFromExtension(Extension extension) {
+        var code = extension.getValue().toString();
+        // validate code membership
+        assert Enumerations.FHIRAllTypes.fromCode(code) != null;
+        return new CodeDef(MeasureConstants.POPULATION_BASIS_URL, code);
     }
 
-    private boolean isGroupIncreaseImprovementNotation(Measure measure, MeasureGroupComponent group) {
-        // default improvement Notation
-        boolean isIncreaseImpNotation = true;
-        boolean useGroupImpNotation = groupHasImprovementNotationExt(group);
-        if (useGroupImpNotation) {
-            isIncreaseImpNotation = isIncreaseImprovementNotation(getGroupImprovementNotationExt(group));
-        } else if (measure.hasImprovementNotation()) {
-            isIncreaseImpNotation = isIncreaseImprovementNotation(measure.getImprovementNotation());
+    public CodeDef getMeasureImprovementNotation(Measure measure) {
+        if (measure.hasImprovementNotation()) {
+            var improvementNotationValue = measure.getImprovementNotation();
+            var codeDef = new CodeDef(
+                    improvementNotationValue.getCodingFirstRep().getSystem(),
+                    improvementNotationValue.getCodingFirstRep().getCode());
+            validateImprovementNotationCode(codeDef);
+            return codeDef;
         }
+        return null;
+    }
+
+    public CodeDef getGroupImpNotation(MeasureGroupComponent group) {
+        var ext = group.getExtensionByUrl(MEASUREREPORT_IMPROVEMENT_NOTATION_EXTENSION);
+        if (ext != null) {
+            var value = ext.getValue();
+            assert value instanceof CodeableConcept;
```

Comment:
assert is disabled by default by Java and must be enabled explicitly.  I recommend either conditional logic and throwing an explicit Exception, or using a library like Guava's Preconditions.